src/all/koharu/src/eu/kanade/tachiyomi/extension/all/koharu/Koharu.kt

```diff
@@ -62,9 +62,22 @@ class Koharu(
 
     private fun remadd() = preferences.getBoolean(PREF_REM_ADD, false)
 
-    override fun headersBuilder() = super.headersBuilder()
-        .add("Referer", "$baseUrl/")
-        .add("Origin", baseUrl)
+    private fun getDomain(): String {
+        try {
+            val host = client.newCall(GET(baseUrl, headers)).execute().request.url.host

```

Comment:
disable redirect on a new client and use that, get url from the `location` header